### PR TITLE
Improve nx-btn accessibility RSC-398

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-398

Added tooltips to icon only buttons, added examples of "normal" buttons with only icons as content. Updated/improved descriptions and examples.

### CODE REVIEW

File: gallery/src/components/NxButton/NxButtonIconExample.tsx [hunk 1]
```diff
@@ -5,16 +5,24 @@
...
-import {faSync} from '@fortawesome/free-solid-svg-icons';
+import { faSync, faSave, faEdit } from '@fortawesome/free-solid-svg-icons';
 
-import { NxButton, NxFontAwesomeIcon } from '@sonatype/react-shared-components';
+import { NxButton, NxFontAwesomeIcon, NxTooltip } from '@sonatype/react-shared-components';
 
 const NxButtonIconExample = () =>
   <div className="nx-btn-bar">
-    <NxButton>
+    <NxButton aria-label="sync">
```
Comment (rpokorny) [inline gallery/src/components/NxButton/NxButtonIconExample.tsx:14]: Typically in a button containing both an icon and text, the icon is semantically redundant and should not have a label.

File: gallery/src/components/NxButton/NxButtonIconExample.tsx [hunk 2]
```diff
@@ -5,16 +5,24 @@
...
       <NxFontAwesomeIcon icon={faSync}/>
       <span>Icons in buttons</span>
     </NxButton>
+
+    <NxTooltip title="Save">
```
Comment (rpokorny) [inline gallery/src/components/NxButton/NxButtonIconExample.tsx:19]: I thought the idea with this ticket was to create an `NxIconOnlyButton` component that encapsulated the need for a tooltip.

**Comment by rpokorny** [comment]
Are there zeplin mockups that you can link to for the now-multiple kinds of icon-only buttons?

File: gallery/src/components/NxButton/NxButtonPage.tsx [hunk 3]
```diff
@@ -68,15 +68,18 @@ export default function NxButtonPage() {
                           id="nx-button-icon-example"
                           liveExample={NxButtonIconExample}
                           codeExamples={nxButtonIconCode}>
-        An example of a button containing an icon in addition to text.
+        An example of buttons containing icons. When an icon's only content consists of an icon it should be wrapped
+        in an <code className="nx-code">NxTooltip</code> and its <code className="nx-code">aria-label</code> set.
```
Comment (rpokorny) [inline gallery/src/components/NxButton/NxButtonPage.tsx:72]: This isn't true.  If the button has a tooltip, you shouldn't need to set the aria-label on anything.

File: gallery/src/components/NxButton/NxButtonPage.tsx [hunk 4]
```diff
@@ -68,15 +68,18 @@ export default function NxButtonPage() {
...
       <GalleryExampleTile title="Icon only buttons"
                           id="nx-button-icon-only-example"
                           liveExample={NxButtonIconOnlyExample}
                           codeExamples={nxButtonIconOnlyCode}>
-        An example of a button containing only an icon. For accessibility purposes it is important to add an
-        aria-label for the screen reader to interpret the content correctly.
+        The <code className="nx-code">icon-only</code> variant is meant to be used in places like lists or tables where
+        a slightly smaller button size is desirable. For accessibility purposes it is important to wrap it in
+        an <code className="nx-code">NxTooltip</code> and an <code className="nx-code">aria-label</code> for the screen
+        reader to interpret the content correctly.
```
Comment (rpokorny) [inline gallery/src/components/NxButton/NxButtonPage.tsx:82]: I think it's confusing to have two different kinds of "icon only buttons" only one of which is the `icon-only` variant.  This should be renamed in a way that reflects it's intended usage.

Comment (glenhunter) [reply gallery/src/components/NxButton/NxButtonPage.tsx:82]: > I think it's confusing to have two different kinds of "icon only buttons" only one of which is the `icon-only` variant. This should be renamed in a way that reflects it's intended usage.

The difference is that one is just a "regular" button whose content is nothing more than an icon, and the `icon-only` variant which has design differences and intended usage that are different. I agree that the naming sucks.

**Comment by glenhunter** [comment]
> Are there zeplin mockups that you can link to for the now-multiple kinds of icon-only buttons?

No. They aren't necessary (at this time @dturner32 might decide to tweak them at some point), the idea was just to provide examples of the regular button variants with content that is just an icon.

**Comment by rpokorny** [comment]
> No. They aren't necessary (at this time @dturner32 might decide to tweak them at some point), the idea was just to provide examples of the regular button variants with content that is just an icon.

I mean, I'm finding them necessary as otherwise I don't know whether or not the left and right padding around the icon is correct.  I suspect it isn't but I can't say for sure without a spec.  Additionally I was hoping the spec would help clarify when exactly one kind of icon-only button is supposed to be used vs the other.

**Comment by glenhunter** [comment]
> > No. They aren't necessary (at this time @dturner32 might decide to tweak them at some point), the idea was just to provide examples of the regular button variants with content that is just an icon.
> 
> I mean, I'm finding them necessary as otherwise I don't know whether or not the left and right padding around the icon is correct. I suspect it isn't but I can't say for sure without a spec. Additionally I was hoping the spec would help clarify when exactly one kind of icon-only button is supposed to be used vs the other.

IIRC the icon-only variant was only intended to be used in lists and tables, that's why it's smaller. However IQ and RM both have some places where there is an icon-only button next to a form field and the smaller `icon-only` variants look bad there because the height of the button and the form fields don't match.

**Comment by glenhunter** [comment]
> > No. They aren't necessary (at this time @dturner32 might decide to tweak them at some point), the idea was just to provide examples of the regular button variants with content that is just an icon.
> 
> I mean, I'm finding them necessary as otherwise I don't know whether or not the left and right padding around the icon is correct. I suspect it isn't but I can't say for sure without a spec. 

My expectation was that they would just follow the spacing of the other buttons.

**Comment by glenhunter** [comment]
Closing until @dturner32 returns and we have better designs and clearer direction.

File: gallery/src/components/NxButton/NxButtonIconExample.tsx (hunk 1 above)
Comment (glenhunter) [reply gallery/src/components/NxButton/NxButtonIconExample.tsx:14]: > Typically in a button containing both an icon and text, the icon is semantically redundant and should not have a label.

https://github.com/sonatype/sonatype-react-shared-components/pull/247/commits/dedb255a7ff9c5c641fd7bfc853d02d008d45534

File: gallery/src/components/NxButton/NxButtonPage.tsx (hunk 3 above)
Comment (glenhunter) [reply gallery/src/components/NxButton/NxButtonPage.tsx]: > This isn't true. If the button has a tooltip, you shouldn't need to set the aria-label on anything.

https://github.com/sonatype/sonatype-react-shared-components/pull/247/commits/e562b598f7b399b6a6f123cac9a02451fbd78726

**Pull request closed without merging**